artifacthub/library/general/requirednamespacequotas/1.0.0/template.yaml

```diff
@@ -0,0 +1,41 @@
+apiVersion: templates.gatekeeper.sh/v1
+kind: ConstraintTemplate
+metadata:
+  name: requirednamespacequota
+  annotations:
+    metadata.gatekeeper.sh/title: "Required Namespace Quota"
+    metadata.gatekeeper.sh/version: 1.0.0
+    description: >-
+      Ensures that all namespaces have a ResourceQuota defined.
+spec:
+  crd:
+    spec:
+      names:
+        kind: RequiredNamespaceQuota
+      validation:
+        openAPIV3Schema:
+          type: object
+          properties:
+            message:
+              type: string
+  targets:
+    - target: admission.k8s.gatekeeper.sh
+      rego: |
+        package requirednamespacequota
+
+        import rego.v1
+
+        # Enforce that every namespace must have a ResourceQuota
+        violation[{"msg": msg}] if {
```

Comment:
Currently v1 syntax is not supported in gatekeeper. We either need to hold off on merging this PR until https://github.com/open-policy-agent/frameworks/pull/517 is merged or use v0 syntax for this ConstraintTemplate.